src/benchmark_mediation.py

```diff
@@ -617,174 +619,309 @@ def alternative_estimator(y, t, m, x, regularization=True):
-        diff_mu_m1[test_index] = y_reg.predict(get_interactions(interaction, x, t1, m1)[test_index, :]) -\
-            y_reg.predict(get_interactions(interaction, x, t0, m1)[test_index, :])
-        diff_mu_m0_reg = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][ind_t0, :], diff_mu_m0[test_index][ind_t0])
-        diff_mu_m1_reg = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][ind_t0, :], diff_mu_m1[test_index][ind_t0])
-        f_00x[test_index] = m_prob.predict_proba(get_interactions(interaction, t0, x)[test_index, :])[:, 0]
-        f_01x[test_index] = m_prob.predict_proba(get_interactions(interaction, t0, x)[test_index, :])[:, 1]
-        theta_0x[test_index] = diff_mu_m0_reg.predict(x[test_index, :]) * f_00x[test_index] +\
-            diff_mu_m1_reg.predict(x[test_index, :]) * f_01x[test_index]
-
-        f_m0x[test_index] = m_prob.predict_proba(get_interactions(interaction, t0, x)[test_index, :])[test_ind, m[test_index]]
-        f_m1x[test_index] = m_prob.predict_proba(get_interactions(interaction, t1, x)[test_index, :])[test_ind, m[test_index]]
-
-        mu_i[test_index] = y_reg.predict(get_interactions(interaction, x, tr, mr)[test_index, :])
-        mu_t1[test_index] = y_reg.predict(get_interactions(interaction, x, t1, mr)[test_index, :])
-        mu_t0[test_index] = y_reg.predict(get_interactions(interaction, x, t0, mr)[test_index, :])
-
-        mu_t1m1[test_index] = y_reg.predict(get_interactions(interaction, x, t1, m1)[test_index, :])
-        mu_t1m0[test_index] = y_reg.predict(get_interactions(interaction, x, t1, m0)[test_index, :])
-        reg_y_t1m1_t0 = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][ind_t0, :], mu_t1m1[test_index][ind_t0])
-        reg_y_t1m0_t0 = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][ind_t0, :], mu_t1m0[test_index][ind_t0])
-        reg_y_t1m1_t1 = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][~ind_t0, :], mu_t1m1[test_index][~ind_t0])
-        reg_y_t1m0_t1 = RidgeCV(alphas=alphas, cv=CV_FOLDS).fit(x[test_index, :][~ind_t0, :], mu_t1m0[test_index][~ind_t0])
-        f_10x[test_index] = m_prob.predict_proba(get_interactions(interaction, t1, x)[test_index, :])[:, 0]
-        f_11x[test_index] = m_prob.predict_proba(get_interactions(interaction, t1, x)[test_index, :])[:, 1]
-
-        psi_0x[test_index] = reg_y_t1m0_t0.predict(x[test_index, :]) *\
-            f_00x[test_index] + reg_y_t1m1_t0.predict(x[test_index, :]) *\
-            f_01x[test_index]
-        psi_1x[test_index] = reg_y_t1m0_t1.predict(x[test_index, :]) *\
-            f_10x[test_index] + reg_y_t1m1_t1.predict(x[test_index, :]) *\
-            f_11x[test_index]
-
+        # predict f(M=m|,T=t,X)
+        f_00x[test_index] = m_prob.predict_proba(
+            get_interactions(interaction, t0, x)[test_index, :]
+        )[:, 0]
+        f_01x[test_index] = m_prob.predict_proba(
+            get_interactions(interaction, t0, x)[test_index, :]
+        )[:, 1]
```

Comment:
Sound like predict_proba is run several times, which is awkward.

Reply:
I don't get it, do you mean it's not readable?

Reply:
No, you should run the predict proba once, store the output in a varaible and then deliver the different chunks into the different variables

Reply:
done, [af8fff6](https://github.com/judithabk6/med_bench/pull/35/commits/af8fff69af8e27ee5aaa0bfd2083f5252add862c)